### PR TITLE
cached messages

### LINKED ISSUE #428: Cached message

### PR DESCRIPTION
**Pull request by l3r8yJ**
closes #428 

@h1alexbel take a look, please

---

## PR-Codex overview
This PR focuses on improving the caching mechanism in the `Timestamped` and `WithPartition` classes. 

### Detailed summary
- Added imports for `Scalar`, `Sticky`, and `Synced`
- Changed the type of the `message` field in `Timestamped` and `WithPartition` to `Scalar<ProducerRecord<K, X>>`
- Updated the constructor of `Timestamped` and `WithPartition` to use `Synced` and `Sticky` for caching
- Added a todo comment to write tests for the caching mechanism in both classes

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by zoeself** [comment]
@l3r8yJ thank you for your Pull Request. I'll assign someone to review it soon. 

 If this PR solves a ``todo`` from the code, please don't forget to remove it.

**Comment by zoeself** [comment]
@h1alexbel please review this Pull Request. Deadline (when it should be merged or closed) is ``2023-08-23T17:59:09.750988``.

You should check if the requirements have been implemented (partially or in full), if there are unit tests covering the changes and if the CI build passes. Feel free to reject the PR or ask for changes if it's too big or not clear enough.

Estimation here is ``30 minutes``, that's how much you will be paid. You will be paid even if this PR gets rejected.

**Review by h1alexbel — APPROVED** [review]

**Comment by h1alexbel** [comment]
@l3r8yJ can we introduce any tests on this?

**Comment by l3r8yJ** [comment]
@h1alexbel I'll create a puzzle

**Comment by l3r8yJ** [comment]
@h1alexbel take a look, please

**Comment by zoeself** [comment]
@h1alexbel Don't forget to close this ticket before the deadline (2023-08-23T17:59:10). You are past the first half of the allowed period.

File: src/main/java/io/github/eocqrs/kafka/data/WithPartition.java [hunk 1]
```diff
@@ -55,7 +64,7 @@ public WithPartition(
     final Message<K, X> msg
   ) {
     this.partition = prtn;
-    this.message = msg;
+    this.message = new Synced<>(new Sticky<>(msg));
```
Comment (h1alexbel) [inline src/main/java/io/github/eocqrs/kafka/data/WithPartition.java:67]: @l3r8yJ why we are doing Sync here?

**Review by h1alexbel — COMMENTED** [review]
looks good to me

File: src/main/java/io/github/eocqrs/kafka/data/WithPartition.java (hunk 1 above)
Comment (l3r8yJ) [reply src/main/java/io/github/eocqrs/kafka/data/WithPartition.java:67]: @h1alexbel Due to the fact that `Sticky` is not thread-safe by default

**Comment by h1alexbel** [comment]
@l3r8yJ thanks

**Comment by zoeself** [comment]
@h1alexbel thank you for resolving this ticket. I've just added it to your active invoice. You can always check all your invoices and more on the [Contributor Dashboard](https://self-xdsd.com).

**Pull request merged**